### PR TITLE
minimize_size: New `repack` option for qemu-img-based repacking

### PR DESCRIPTION
**Pull request by liori**
I haven't actually tested this on vdi, but based on `qemu-img`'s documentation, it should work just as well.

### CODE REVIEW

**Comment by andsens** [comment]
Cool! Would it ever make sense to both use vmware-diskmanager *and* qemu-img? Otherwise I'd suggest making `shrink` an enum with the two options and `True/False` as backwards compat.

**Comment by liori** [comment]
I don't know, I don't have VMWare installed anywhere. I guess there should be little benefit in running them both. I'll make it into an enum then. Do you prefer squashed merges or should I add the changes as a separate commit?

**Comment by andsens** [comment]
Squashed would be best :-)

**Comment by andsens** [comment]
Any update on this?

**Comment by liori** [comment]
Sorry for the delay. The change is now implemented. Thinking about it, I'd make a small refactoring here and split `shrink.AddRequiredCommands` into three separate tasks for `zerofree` and both types of `shrink` — though in a separate commit or push request, later. Is that desirable?

File: bootstrapvz/plugins/minimize_size/README.rst [hunk 1]
```diff
@@ -35,8 +39,13 @@ Settings
    Default: false
    ``optional``
 -  ``shrink``: Whether the volume should be shrunk. This setting works
-   best in conjunction with the zerofree tool.
-   Valid values: true, false
+   best in conjunction with the zerofree tool. Valid values:
+
+   -  false: Do not shrink.
+   -  ``vmware-vdiskmanage`` or true: Shrink using the `vmware-vdiskmanager`
```
Comment (andsens) [inline bootstrapvz/plugins/minimize_size/README.rst]: Woops, typo. Missing `r` in `vmware-vdiskmanager`

File: bootstrapvz/plugins/minimize_size/tasks/shrink.py [hunk 2]
```diff
@@ -8,6 +8,20 @@
 import os
 
 
+def get_shrink_type(plugins):
```
Comment (andsens) [inline bootstrapvz/plugins/minimize_size/tasks/shrink.py]: The tasks module should really only contain actual `Task`s. I recommend putting this in `__init__`

**Comment by andsens** [comment]
A split-up of the required commands task would be appreciated. Then you don't need the helper function either, you can just inline it in `resolve_tasks()`.

**Comment by liori** [comment]
I fixed the typo and added a commit that implements the split of `shrink.AddRequiredCommands`. The helper function is still also used in the `validate_manifest` function, so I didn't inline it; but I moved it to the `minimize_size/__init__.py` file. I'm not sure if the identifiers/description strings for new tasks are proper?

**Comment by andsens** [comment]
Looks great! Merging.

The past tense of "shrink" is "shrunk" btw., not that it matters, it's a temp file after all :-)

**Pull request merged**